src/engine/input.lua

```diff
@@ -72,6 +77,23 @@ Input.gamepad_cursor_size = 10
 Input.gamepad_cursor_x = (love.graphics.getWidth()  / 2) - (Input.gamepad_cursor_size / 2)
 Input.gamepad_cursor_y = (love.graphics.getHeight() / 2) - (Input.gamepad_cursor_size / 2)
 
+Input.mouse_button_max = 3
+Input.mouse_down = {
+    [1] = {0, 0, 0}, -- x, y, presses
```

Comment:
please use keys instead of a comment, also you can have more than three mouse buttons so maybe you can assign the defaults elsewhere

Reply:
as far as i know, there's no way to detect ahead of time how many mouse buttons the user might have; i adjust the default in Input.onMousePressed if the user ever inputs a mouse button higher than the current default